src/Features/Core/Portable/CodeFixes/Qualify/AbstractQualifyMemberAccessCodeFixprovider.cs

```diff
@@ -0,0 +1,47 @@
+﻿// Copyright (c) Microsoft.  All Rights Reserved.  Licensed under the Apache License, Version 2.0.  See License.txt in the project root for license information.
+
+using System.Collections.Immutable;
+using System.Threading.Tasks;
+using Microsoft.CodeAnalysis.Diagnostics;
+using Microsoft.CodeAnalysis.Shared.Extensions;
+using Microsoft.CodeAnalysis.Simplification;
+
+namespace Microsoft.CodeAnalysis.CodeFixes.Qualify
+{
+    internal abstract class AbstractQualifyMemberAccessCodeFixprovider<TSyntaxNode> : CodeFixProvider where TSyntaxNode : SyntaxNode
+    {
+        public sealed override ImmutableArray<string> FixableDiagnosticIds => ImmutableArray.Create(IDEDiagnosticIds.AddThisOrMeDiagnosticId);
+
+        protected abstract SyntaxNode GetReplacement(TSyntaxNode node);
+        protected abstract string GetTitle();
+        protected abstract string GetId();
+
+        public override async Task RegisterCodeFixesAsync(CodeFixContext context)
+        {
+            var document = context.Document;
+            var span = context.Span;
+            var cancellationToken = context.CancellationToken;
+
+            var root = await document.GetSyntaxRootAsync(cancellationToken).ConfigureAwait(false);
+            var model = await document.GetSemanticModelAsync(cancellationToken).ConfigureAwait(false);
+
+            var token = root.FindToken(span.Start, findInsideTrivia: true);
+            if (!token.Span.IntersectsWith(span))
+            {
+                return;
+            }
+
+            var node = token.GetAncestor<TSyntaxNode>();
+            if (node == null)
+            {
+                return;
+            }
+
+            var codeAction = new QualifyMemberAccessCodeAction(
+                GetTitle(),
+                c => document.ReplaceNodeAsync(node, GetReplacement(node), c),
```

Comment:
I can go either way on this.  The only thing that `GetReplacement()` does is add a `this.` prefix, and by definition of this style rule, that can't get simplified, but I can also argue that if the simplifier ever gets more complicated (see what I did there?) that this would be necessary.

Would there be a noticeable perf hit in adding a simplifier annotation that we know will result in a noop?